packages/design-system-old/src/utils/icon-loadables.tsx

```diff
@@ -0,0 +1,43 @@
+// IMPORTANT: Please keep this file in sync with packages/design-system/src/Icon/loadables.tsx.
+
+import React from "react";
+import loadable from "@loadable/component";
+import {
+  RemixiconReactIconProps,
+  RemixiconReactIconComponentType,
+} from "remixicon-react";
+
+function importIconImpl(
+  importFn: () => Promise<any>,
+  options?: {
+    resolveComponent: (m: any) => React.ComponentType;
+  },
+) {
+  const IconLoader = loadable.lib(importFn, options as any);
+
+  return function AppsmithLoadable(props: any) {
+    return (
+      // Using `<svg />` as a loading fallback – to make sure sizes set by <IconWrapper>
+      // apply even while the full icon is still loading.
+      // And passing all props – if the parent component sets `width` or `height`,
+      // we want to make sure it’s applied to the loading fallback as well.
+      <IconLoader fallback={<svg {...props} />}>
+        {(Icon: React.ComponentType) => <Icon {...props} />}
+      </IconLoader>
+    );
+  };
+}
```

Comment:
Note: using a bunch of `any`s here instead of correct typing because `loadable.lib()` types aren’t fully correct. Specifically:
 - `loadable.lib()` is typed as if it doesn’t support [the `resolveComponent` option](https://loadable-components.com/docs/api-loadable-component/#loadablelib) – whereas in practice (and per the docs) it does.
 - the render callback (the `(Icon: React.ComponentType) => <Icon {...props} />` one) is typed to return an object that’s shaped like `{ default: React.ComponentType }` – whereas in practice, it returns the component itself.